packages/gitbook/src/lib/links.ts

```diff
@@ -45,6 +45,11 @@ export async function getBasePath(): Promise<string> {
 export async function getHost(): Promise<string> {
     assertIsNotV2();
     const headersList = await headers();
+    const mode = headersList.get('x-gitbook-mode');
+    if (mode === 'proxy') {
+        return headersList.get('x-forwarded-host') ?? '';
```

Comment:
We can't rely on `x-gitbook-host` for `proxy` mode because that value is overwritten by our content worker.

Instead we rely on the `forwarded-host`  because it's using the `inputURL` host https://github.com/GitbookIO/gitbook/blob/685325cdc7e5ef0b9d4f8a85b98a5b658ef8aec2/packages/gitbook/src/middleware.ts#L174

Reply:
But why is the request going through the content worker?

Why not directly make the request to `open.gitbook.com` ?

Reply:
Because of caching in content worker